### PR TITLE
Add dd tracing to opensearch

### PR DESCRIPTION
**Pull request by michaelginalick**
## Problem

Trace opensearch requests

## Solution

Add datadog tracing library and wrap the opensearch client

### CODE REVIEW

File: opensearchconfig.go [hunk 1]
```diff
@@ -69,6 +70,8 @@ func GetConfig() (Config, error) {
 // ConfigFromEnv creates an OpenSearch config object from environment variables
 func ConfigFromEnv(ctx context.Context) (opensearch.Config, error) {
 	opensearchConfig := opensearch.Config{}
+	// Add datadog tracing
+	opensearchConfig.Transport = httptrace.WrapRoundTripper(&http.Transport{})
```
Comment (mike-carey) [inline opensearchconfig.go:74]:
```suggestion
	opensearchConfig.Transport = &http.Transport{}
```

**Review by mike-carey — CHANGES_REQUESTED** [review]
Can we please make this configurable via the `DD_TRACE_ENABLED` environment variable?  This is a public repo and not everyone may be using DataDog.

File: opensearchconfig.go [hunk 2]
```diff
@@ -77,9 +80,9 @@ func ConfigFromEnv(ctx context.Context) (opensearch.Config, error) {
 
 	if cfg.SkipSSL {
 		log.Debug("skipping ssl")
-		opensearchConfig.Transport = &http.Transport{
+		opensearchConfig.Transport = httptrace.WrapRoundTripper(&http.Transport{
 			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
+		})
 	}
```
[the remark this answers is not in the document]
Comment (mike-carey) [reply opensearchconfig.go:87]: You will need to import "os" if you accept this.

File: opensearchconfig.go [hunk 3]
```diff
@@ -22,6 +24,8 @@ const (
 	AuthBasic = "basic"
 	// AuthIAM is the auth type to use if not using the internal user database on the cluster
 	AuthIAM = "iam"
+	// ENV variable to controll whether traces are send to datadog
```
Comment (mike-carey) [inline opensearchconfig.go:27]:
```suggestion
	// ENV variable to control whether traces are send to datadog
```

File: opensearchconfig.go [hunk 4]
```diff
@@ -69,6 +73,8 @@ func GetConfig() (Config, error) {
 // ConfigFromEnv creates an OpenSearch config object from environment variables
 func ConfigFromEnv(ctx context.Context) (opensearch.Config, error) {
 	opensearchConfig := opensearch.Config{}
+	// Add datadog tracing
```
Comment (mike-carey) [inline opensearchconfig.go:76]:
```suggestion
```

**Review by mike-carey — APPROVED** [review]

**Review by andrewbatallones — APPROVED** [review]

**Pull request merged**